acp/internal/controller/agent/agent_controller_test.go

```diff
@@ -16,104 +17,82 @@ import (
 )
 
 var _ = Describe("Agent Controller", func() {
-	Context("When reconciling a resource", func() {
-		const resourceName = "test-agent"
-		const llmName = "test-llm"
-		const humanContactChannelName = "test-humancontactchannel"
-
-		ctx := context.Background()
-
-		typeNamespacedName := types.NamespacedName{
-			Name:      resourceName,
-			Namespace: "default",
-		}
-
-		BeforeEach(func() {
-			// Create a test LLM
-			llm := &acp.LLM{
-				ObjectMeta: metav1.ObjectMeta{
-					Name:      llmName,
-					Namespace: "default",
-				},
-				Spec: acp.LLMSpec{
-					Provider: "openai",
-					APIKeyFrom: &acp.APIKeySource{
-						SecretKeyRef: acp.SecretKeyRef{
-							Name: "test-secret",
-							Key:  "api-key",
-						},
+	const resourceName = "test-agent"
+	const llmName = "test-llm"
+
+	ctx := context.Background()
+
+	typeNamespacedName := types.NamespacedName{
+		Name:      resourceName,
+		Namespace: "default",
+	}
+
+	BeforeEach(func() {
+		// Create a test LLM
+		llm := &acp.LLM{
```

Comment:
What was the reason for not using `TestLLM` from `utils`?

Reply:
oh i tried to refactor this test to follow the patterns better, looks like it didn't pick up the "use utils" instructions. 

anyhow fine to improve this more later

Reply:
fixed